### DISCUSSION TITLE
discussion

### DISCUSSION THREAD

**Discussion by icsabai** [csabaiBio/elte_ml_journal_club#93] [Ideas]
Mivel nincs jobb ML email csoport, nyitok itt egy általános diszkusszió/kéredések streamet.

Most beszéltünk Dobos Lacival, hogy a (sima) photo-z esetén lehetne-e neuronhálós módszernél is alkalmazni olyasmit, amit végül az SDSS photo-z-nél használtunk. Ott a k-nearest neighbor-re csinálunk egy lineáris regressziót, ami nyilván jobb, mint egy globális regresszió, meg jobb a k-nearest átlagnál is, mert elvileg ez tud extrapolálni is. A neuronhálónál olyasmit lehetne, hogy a bemeneten megkapja a k-nearest neighbor magnitúdóit és redshiftjét is, meg a jóslandónak a magnitúdóit is, és úgy tanul, és úgy is jósol.  Lehet,  hogy a tanító halmazon (nem a lokális szomszédokon) túli extrapoláció így nem jönne létre, de a lineáris regressziónál lehetne robusztusabb.

Egy kérdés ami egyből felmerült, hogy a k-nearest neighbor sorrendjére ennek nem kellene érzékenynek lenni, azaz talán valami permutation-equivariant network kellene, a gráf konvolúciós hálók kapcsán Andris nézegetett hasonlókat. Ugyanez a kérdés merül fel másik problémánál, amikor pl. LIDAR adatok esetén pontfelhők pontjainak koordinátáiból kellene tanulni, itt is mindegy a sorrend.

**Comment by icsabai** [comment]
A New Scientistben volt ez a [cikik](https://kooplex-fiek.elte.hu/seafile/f/c8e5a704a81c4da89502/) ezzel a címmel: __Companies could be fined if they fail to explain decisions made by AI__. [Itt van](https://ico.org.uk/about-the-ico/ico-and-stakeholder-consultations/ico-and-the-turing-consultation-on-explaining-ai-decisions-guidance/) a "nemzeti konzultáció" ami alapján íródott. Az "explain mammo" cikkhez esetleg meg is lehetne hivatkozni, illetve lehet, hogy vannak benne tartalmas szövegek.

**Comment by icsabai** [comment]
Most kaptam ezt a machine learning viccet, ha valaki nem ismerné. Ezzel kívánok mindenkinek Kellemes Ünnepeket!

https://www.reddit.com/r/ProgrammerHumor/comments/ed82as/how_to_get_a_job_as_a_machine_learning_engineer/

**Comment by B-Lorentz** [comment]
Sziasztok!
Ma beszéltünk a [ReZero](https://arxiv.org/pdf/2003.04887v1.pdf)-ról. Fogtam a [PyTorch Transformer tutorialját](https://pytorch.org/tutorials/beginner/transformer_tutorial.html) (ebben egy viszonylag kicsi Transformert tanítanak wikipedia-szövegen), és minden további változtatás nélkül (majdnem, a szélességet 256-ra növeltem) beletettem a ReZero-t (egész TransformerEncoder layereket csomagoltam bele, tehát az Attention-t és a Dense réteget együtt, míg az eredeti cikkben mindkettőnek külön alfája van). Az eredmény nem annyira extrém mint a cikkben, de azért markánsan segít a nagyobb (8 rétegű) hálónak:
![rezero](https://user-images.githubusercontent.com/44694582/77685126-3c452580-6f9b-11ea-871a-370446beaa44.png)

**Comment by icsabai** [comment]
Ügyes! Ha jól értem ez a training loss. Mi történik közben a teszttel?

**Comment by B-Lorentz** [comment]
> Ügyes! Ha jól értem ez a training loss. Mi történik közben a teszttel?

A négy végső test-loss:
Shallow ReZero: 5.357
Shallow Transformer: 5.383
Deep ReZero: 5.20,
Deep Transformer: 6.74

**Comment by B-Lorentz** [comment]
Sziasztok!
Gondolkoztam/körbenéztem a hipergráf modellekből való térépítésen:
-Wolframéknak van egy fórumuk, ahol például szó esik arról, hogy milyennek kellene lennie egy gráfnak, hogy az határesetben (messziről nézve) egy izotróp térnek feleljen meg, például a lépészám-távolság az euklideszi távolsághoz tartson, ne maradjon mindig manhattan-távolság (mint egy akármilyen nagy sima rácson). Egyszerű  javaslatként egy felhasználó több különböző irányú rács összeépítését javasolja (csak rajzban, nem update-rule-okkal) https://community.wolfram.com/groups/-/m/t/1945186

-Elvileg biztos, hogy lehetséges kellene legyen "tömör tér" felépítése a megfelelő kezdőfeltételekkel. Megmutatják, hogy a hipergráf-modellben elég közvetlenül lehet Turing-gépet is építeni https://www.wolframphysics.org/technical-introduction/equivalence-and-computation-in-our-models/correspondence-with-other-systems/ Szóval legrosszabb esetben egy programozott "térépítő" lépésenként és élenként összeépítheti a kivánt teret. De valahogy ezt mégsem érezzük a fizikai valóság hihető modelljének :). De a self-loop hiperélek (amelyek ugyanabba a csúcsba térnek vissza, ahonnan jönnek) nagyon megkönnyítik, hogy akármilyen információt eltároljunk a gráfban, "a tulajdonléppeni szerkeztének" megzavarása nélkül. Szinte csalásnak érződik...

**Comment by icsabai** [comment]
Ráadásul én egy olyat szeretnék a "táguló tér" mintájára, ami belülről folyamatosan növekszik, nem a szélén. Így Wolframéknál ez https://www.wolframphysics.org/technical-introduction/typical-behaviors/rules-depending-on-two-ternary-relations-the-23-33-case/img/rules-depending-on-two-ternary-relations-the-2_3---3_3-case_io_21.en.png nem jó, mert ez csak úgy növeli, mint a horgolás.

Bocs, hogy tegnap összevissza próbáltam mutogatni, itt https://kooplex.vo.elte.hu/ownCloud/s/a1LyuXYPlDKd03I van Beck Robi diplomamunkája, ez jó összefoglaló, meg vannak benne hivatkozások. A CDT-ben mint mondtam 2-3-4D tetraéderek vannak, de Wolframnál is a "hipergráf" az kb. háromszögeket jelent. A tetrahedrált cellázás dulálisa (mint Voronoi vs. Delaunay) gráf:
![image](https://user-images.githubusercontent.com/18164480/80226039-297e4900-864c-11ea-8b8d-76f2bfc39f6c.png)

A CDT-nél sztochasztikus dinamika van, azaz nem sejtautomata, mert két szomszédos cella nem tud egymástól függetlenül változni. Ezen gondolkoztam sokat, és valami olyasmi lehetne a megoldás, hogy színezni kell a gráfot (Tait coloring) és ciklikusan a 3 szín updatelődik.

Úgy mehetne a tér tágulás, hogy olyan szabály lenne, hogy bizonyos toplógiai hibáknál (wathever it means) beszúródnának új csúcsok/tetraéderek, máshol meg épp eltűnnének (gravitáció), illetve a fenti ábrán az alsó transzformációval kisímulhatna/tovaterjedne.  Csak mindezt ténylegesen nem sikerült még leimplementálni, hogy pl. tudja a Kepler törvényeket. Habár szerintem a Kepler mozgás lesz az egyszerűbb, az egyenes vonalú egyenletes mozgásra még kevesebb ötletem van. Az a sejtésem kevésbé fundamentális mozgás mint amennyire ezzel indul a fizika könyv. De persze ezek csak spekulációk.

**Comment by B-Lorentz** [comment]
Elolvastam a diplomamunkát, nagyon érdekes. Egyébként szerintem a kötögető modellek elvileg nem lehetetlenek, mivel amíg a "kötőtű" máshol jár, a megfigyelő nem léptetődik, és így "nem veszi észre", hogy nála áll az idő. De persze egy mindenhol egyszerre, lokálisan működő megoldás sokkal egyszerűb és elegánsabb.

**Comment by icsabai** [comment]
A Wolfram Physics témához, itt egy interjú vele az Edge-n: https://www.edge.org/conversation/stephen_wolfram-computation-all-the-way-down
Sok új nincs benne, de ha valaki inkább hallgatni akarja, mint elolvasni.

**Comment by icsabai** [comment]
Hova vezethet a deep learning:

![future](https://user-images.githubusercontent.com/18164480/92209416-3e39db80-ee8d-11ea-93b8-780846965538.jpg)

**Comment by icsabai** [comment]
Múltkor távoli asszociációként szóba jött a Numerai, ahhoz a "novelty/curiosity driven" alapú reinforcement learninghez kapcsolódóan, amit @B-Lorentz  mondott el. Most jött tőlük a legújabb kiáltványuk: https://medium.com/numerai/building-the-last-hedge-fund-introducing-numerai-signals-12de26dfa69c
A közepe táján írja a "predictive value after being neutralized by all other signals" koncepciót, azaz, hogy olyan _értékes_ dolgot kell kitalálni, ami _ortogonális_ arra amit mások kitaláltak.

**Comment by patbaa** [comment]
Többnyire cikkekről beszélünk itt, amik nagyrészt új módszereket mutatnak be, néha pedig ezek alkalmazásáról. Az NVIDIA leigazolta a Kaggle TOP versenyzők jó részét (vajon főállásban Kaggleznek?). 
Nemrég volt a Kaggle-n egy melanoma detection verseny, itt egy interjú van a győztesekkel:
https://youtu.be/L1QKTPb6V_I?t=2

Ha valaki próbálkozik ML versenyekkel, érdemes végignézni, hogy a Kaggle profik szerint mi fontos, és hogy néz ki egy megoldás, ami 3000 résztvevő közül első helyet éri el. Cikkekben ilyeneket ritkán írnak le.

**Comment by icsabai** [comment]
Ha csütörtökön összefoglalod, nem kell végignéznem a videot :-)

**Comment by patbaa** [comment]
rendben, röviden beszélek róla! esetleg valamelyik alkalommal beszélhetnénk az RA challengeről is, ez zárt körű előadás, a publication embargo nem számít így.

**Comment by icsabai** [comment]
szeminárium most

**Comment by B-Lorentz** [comment]
Egy érdekes fejlemény a technikai/szoftver oldalon: Miután a TF2 "bekonvergált" a PyTorch-hoz és google a jax-al "vágott vissza", a Torch is felszedett egy jax-szerű feature-t: 

https://pytorch.org/docs/master/fx.html

A t.nn.Module-okat placeholder objektumokkal trace-oló és transzformáló funkcionalitás. 

(hogy mondják ezt magyarul? "Helyettesítő objektumokkal a végrehajtást nyomonkövetésének és átlalakításának képessége"?)

**Comment by icsabai** [comment]
Gondolom mindenki látta, hogy kijött az alphafold2 cikk:

Tunyasuvunakool, K., Adler, J., Wu, Z. et al. Highly accurate protein structure prediction for the human proteome. Nature (2021). https://doi.org/10.1038/s41586-021-03828-1

https://www.nature.com/articles/s41586-021-03828-1

Nem csak a módszert írták le, hanem az egész humán proteome-ra lefuttatták és közzétették az eredményeket. Maga a módszer a supplementary-ban van, elég alapos és részletes (62 oldal, 32 számozott algoritmus): 

https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-021-03819-2/MediaObjects/41586_2021_3819_MOESM1_ESM.pdf

Szeretnénk majd tartani róla egy nagyobb lélegzetű szemináriumot. Már találtam szakembert aki a fehérjeszerkezet vizsgálat oldalról átnézi a cikket.  Várok jelentkezőket, akik hajlandóak lennének átnézni a neuronhálós oldalát. 
 
Még nem néztem utána, de bizonyára készítenek róla mások is összefoglalókat, az segíthet az áttekintésben.

**Comment by masterdesky** [comment]
Nem láttam sajnos eddig még előzményt hozzá, így gondoltam ide írok. Az OpenAI "megjelentetett" egy új programozási nyelvet, a Tritont, amire azt ígérik, hogy:

"[Triton 1.0 is] an open-source Python-like programming language which enables researchers with no CUDA experience to write highly efficient GPU code—most of the time on par with what an expert would be able to produce."

Ha jól értem, akkor a CUDA-t egyenesen lecserélő nyelvet akarnak ebből csinálni, open source fejelsztéssel. Valaki esetleg tud erről valamit? Nekem eddig az jött le, hogy érdemes lehet részletesebben utánanézni ennek.

Néhány bővebb infó ezen a linken érhető el, pár példakóddal és a dokumentációval egyetemben: https://www.openai.com/blog/triton/
Itt pedig a GitHub repó hozzá, ahonnan az első verzió beszerezhető: https://github.com/openai/triton

**Comment by icsabai** [comment]
Akit érint: a journal club meg van hirdetve felvehető MSc ( dsadvmachlf20sx ) és PhD (FIZ/3/092) tárgyként. Várhatóan az Alex által már hirdetett "Geometric deep learning" témát kell a kreditért feldolgozni.

Jelzem azt is, hogy áttérünk angol nyelvre, mert magyarul nem beszélők is csatlakoznak a szemináriumhoz.

**Comment by icsabai** [comment]
_The New Scientist magazine had this [story](https://www.newscientist.com/article/2291586-ai-is-helping-us-air-force-to-decide-which-targets-to-strike/) today. I let you to decide if this is the dark or bright side of AI._


The **US military has started using AI to guide its air strikes**, according to Secretary of the Air Force Frank Kendall.

Speaking at the Air Force Association’s Air, Space & Cyber Conference in National Harbor, Maryland, on 20 September, Kendall said that the US Air Force (USAF) had recently “**deployed AI algorithms for the first time to a live operational kill chain**”. He didn’t give details of the strike, such as whether it was by a drone or piloted aircraft, or if there were civilian casualties.

Kendall said this use represented “**moving from experimentation to real military capability**”.


The kill chain is the process of gathering intelligence, analysing and evaluating it and directing weapons to destroy a target. The AI was incorporated into the Air Force Distributed Common Ground System (DCGS), which combines data from various sources, including MQ-9 Reaper drones.

DCGS collects and analyses over 1200 hours of video per day and the USAF is seeking to automate parts of this laborious process, for example with automated target recognition. Human analysts can also make deadly mistakes. In one recent case, a car in Kabul, Afghanistan, supposedly full of explosives was targeted. The vehicle was actually loaded with containers of water, and the resulting drone strike killed 10 innocent people, including seven children.

With the recent withdrawal of US troops from Afghanistan, President Joe Biden announced the US will rely on more “over-the-horizon” operations in which strikes are carried out without forces on the ground to confirm targets. While the new AI can’t order a strike, the fact that it is identifying possible targets for human approval represents a significant inroad into the decision-making process, and the lack of details on the system makes it difficult for experts to evaluate. The USAF told New Scientist it wouldn’t be able to comment before publication.

“Originally, automated target recognition was about identifying and tracking well-defined military targets,” says Jack McDonald, a lecturer in war studies at King’s College London. “We don’t know what kinds of objects or patterns of activity these systems are now trying to recognise.”

Arthur Holland at the United Nations Institute for Disarmament Research in Geneva, Switzerland, says that although expected, use of AI for targeting raises serious policy questions.

“How does one test such algorithms and validate that they are reliable enough to be used in life-or-death missions?” asks Holland. “How can operators know when to trust the AI system and when to override it? If an erroneous strike is launched in part on the basis of an erroneous AI system’s output, does that affect the balance of human responsibility and accountability?”

Holland says these are tricky, but urgent, questions. While there has been much attention given to a possible future in which AI routinely controls “killer robots”, the involvement of AI in lethal military operations now is just as important.

McDonald says that the technology eventually becoming accessible more widely, as has happened with drones, is also a concern. “I’m much more worried about if or when this sort of technology gets in the hands of militias and non-state actors,” he says.

**Comment by icsabai** [comment]
From the news, again, loosely related to the dangers of AI algorithms. Though in this case this is mostly about human decision (most probably based on insight from big data analysis): Frances Haugen's whistleblowing on Facebook's policy. A 
[concise video summary](https://www.bbc.com/news/technology-58784615) of the problem. In her words: _"If Facebook change the algorithm to be safer... they'll make less money"_

**Comment by icsabai** [comment]
**Moore's law for NLP models**

New Scientist has an article on vary large ML models: https://www.newscientist.com/article/mg25133550-800-supersized-ais-are-truly-intelligent-machines-just-a-matter-of-scale/ 
Nothing really new, but they have a figure, plotting the exponential growth of the number of parameters of NLP models. I have copied it and added some stuff (in red), did the "by-eye" extrapolation. If the trend continues, in less than two years it will reach human capacity. At least in "number of synapses".
 
![NLP_MooresLaw](https://user-images.githubusercontent.com/18164480/136454836-e9460e2a-e7cd-4377-89c9-c1ef983df65b.png)

**Comment by icsabai** [comment]
New record in NLP network size: Megatron-Turing 530 billion parameters. Trained on 560x8 A100 GPUs, 340 billion token dataset, for 1 month. Queries need "only" 40 GPUs and 1-2 seconds.

https://developer.nvidia.com/blog/using-deepspeed-and-megatron-to-train-megatron-turing-nlg-530b-the-worlds-largest-and-most-powerful-generative-language-model/

**Comment by icsabai** [comment]
DeepMind has solved the most complex board game Go, then the "holy grail" of molecular biology, protein folding.

Maybe now nuclear fusion, too?:

Degrave, J., Felici, F., Buchli, J. et al. Magnetic control of tokamak plasmas through deep reinforcement learning. Nature 602, 414–419 (2022). [https://doi.org/10.1038/s41586-021-04301-9](https://www.nature.com/articles/s41586-021-04301-9)

What next?

**Comment by icsabai** [comment]
Link for the seminar: https://meet.google.com/cgp-jnvf-rza

**Comment by icsabai** [comment]
__Self-driving human__

This is about an "invention" that came to my mind this morning. As I was commuting to the University I saw all the people on the metro with their mobiles in their hands, earbuds in their ears, browsing news, social media, watching movies, playing games. As the old adage says: time is money, so it is understandable that they do not want to waste their most precious resource on just changing coordinates in space. The problem comes when you need to change to tram, you need to climb stairs, avoid of bumping into other commuters. 

Home office or moving the whole world into VR is a solution, but for a while most probably people want/need to move physically. How cool would it be to do so completely without wasting your attention on it. Same as with self-driving cars. A couple of months ago I have already seen a demo video on self-driving bicycle. You may think that this is useless but if you go to work on bike and you have am evening off it may be good to send the bike home alone. Bikes are not compatible with your cocktail dress or tuxedo, not mentioning the late night drinks. BTW, have you seen the bike park at the Copenhagen railway station .. ? . So, the next step is self-walking humans. I see two ways for implementation. One is a lightweight exoskeleton. This comes with the bonus of faster pace and stronger muscles without being tired. The disadvantage is that this is still a device that has weight, volume, battery time etc. The other solution for the bravehearts is an interface in your spinal cord or brain. With a software update it can be extended from walking to other mechanical duties, like making your bed, washing the dishes, etc. while you are watching that romantic movie or killing zombies.

Do not forget me, if you patent this idea and build a multi-billion company on it :-)

(You are also welcome to pollute this forum with questions, news or (crazy) ideas)

**Comment by icsabai** [comment]
Text-to-image generation like [DALLE-2](https://openai.com/dall-e-2/) or [Imagen](https://imagen.research.google/) are so last-month ... Create a whole [video](https://makeavideo.studio/) from a text prompt.  There is a [paper](https://makeavideo.studio/Make-A-Video.pdf) also that someone may want to review and present.

**Reply by bbeatrix** [reply]
Students (@Andrasvalogatas, @rsarugaII, @Dudgit, @gebhaard, @borbende, @j-a-yoo, @gd4ni96, @tonks8811, @SSzende ), this is a great opportunity for presentation, please volunteer to present this one until next week, or I'll have to start to randomly assign you to different papers...

**Comment by bbeatrix** [comment]
If somebody is planning to present a diffusion model or related papers, these are two great introductory blog posts on the topic, maybe one can include some parts of these in the presentation: 
https://lilianweng.github.io/posts/2021-07-11-diffusion-models/ 
https://benanne.github.io/2022/01/31/diffusion.html

If someone prefers to watch videos instead of reading: 
https://www.youtube.com/watch?v=y7J6sSO1k50
https://www.youtube.com/watch?v=W-O7AZNzbzQ

**Comment by icsabai** [comment]
Why use artificial neurons, if real ones can do the job:

__How Neurons in a Dish Learned to Play “Pong”__
The DishBrain system can send and receive electrophysiological signals to and from living neurons, training the cells to accomplish a task.

https://www.the-scientist.com/news-opinion/how-neurons-in-a-dish-learned-to-play-pong-70613

**Comment by icsabai** [comment]
We do not have to wait another 50 years to beat matrix multiplication efficiency record of AlphaTensor (presented at last seminar) at least for the 5x5 matrices:

https://arxiv.org/abs/2210.04045

_In response to a recent Nature article which announced an algorithm for multiplying 5×5-matrices over Z2 with only 96 multiplications, two fewer than the previous record, we present an algorithm that does the job with only 95 multiplications._ 

It is fun to open the PDF and read the paper from the 2nd page :-)

**Comment by rsarugaII** [comment]
Hello,
I would be happy to give a presentation on "How neurons learned to play
ping pong in a bowl" at the next free slot, which I think will be on 03.11.

Br.
Rudolf Saruga




Istvan Csabai ***@***.***> ezt írta (időpont: 2022. okt. 13.,
Cs, 22:51):

**Comment by icsabai** [comment]
__Lawsuit against copilot__
I guess most of you have heard about OpenAI+Microsoft's [Copilot](https://github.com/features/copilot). In essence it does as all human programmers: if you need a piece of algorithm, you look it up at stackoverflow or other places on the net, copy-paste, adjust variable names etc. This is automatized in Copilot, you just write in as a comment line what you want and AI do the rest. I have not used it seriously, but many say that it is really good.

Now come the lawyers. As they use "people's intellectual results" (to be specific, code on GitHub, that is now owned by Microsoft) this is against certain rights. All grabbed code should be properly acknowledged, maybe ask permit from the code creators, etc. 

New Scientist has an [article](https://www.newscientist.com/article/2346217-microsofts-copilot-code-tool-faces-the-first-big-ai-copyright-lawsuit/) on it, that you cannot read because of their copyright :-) but as I am a subscriber, I lawfully [share](https://kooplex-fiek.elte.hu/seafile/f/deb683dd94724d9db49b/) with you. (Of course not allowing further sharing. Be sure not to do so, this message is now on GitHub and they read it and watch you).

I some sense this is part of the general problem of properly handling/acknowledging the training set, and the learned information in all machine learning, where we use other's data.  

The problem is real but it is hard to find a good solution. I expect GDPR like popups or similar solutions.

**Comment by icsabai** [comment]
Someone shared this with me on the other list (in Hungarian. You can ask AI to translate :-)

Ezen kívül ez szerintem nagyon megdöbbentő: https://www.engraved.blog/building-a-virtual-machine-inside/
A ChatGPT-nek meg lehet mondani, hogy válaszoljon úgy, mintha egy linuxot futtató virtuális gép lenne, ami után lehet fájlokat készíteni benne, stb. Sőt az "internetre" is fel lehet menni vele, ahol aztán a szerző megtalálta a ChatGPT weboldalán az eredeti ChatGPT-t (ami ott már magát Assistant-nak hívja), amit utána megkért, hogy viselkedjen úgy, mintha egy linuxot futtató virtuális gép lenne. Szerintem ez nagyon meta :) .

My reaction was:

Nem lehet, hogy ez a linux terminál ChatGPT vagy egy beugratás, vagy pedig tényleg tud egy valódi linuxot futtatni, rendes internet kapcsolattal? Elég hihetetlen, hogy egy "rendes" nyelvi modell ilyeneket tudjon "fejből". Számítási kapacitás mondjuk van elég a gépben amin fut, és ha úgy vesszük a copilot is tud kódokat írni, a teljes github ismeretében annyira nem is csoda. Meg hát a python meg bash is csak egy nyelv, egyszerűbb, mint az angol. Egyszerű kódokat én is végig tudok léptetgetni és kitalálni az output-ot, szoktak is ilyet feladni vizsgán. A szerver meg kicsit gyorsabb mint mi. Szóval, lehet, hogy mégse hihetetlen. Akkor meg hátborzongató.

**Comment by icsabai** [comment]
Humans strike back: https://arstechnica.com/information-technology/2023/02/man-beats-machine-at-go-in-human-victory-over-ai/

**Comment by icsabai** [comment]
With some of the group we have discussed Christian Szegedy's recent talk on autoformalization at [AI&AUT Expo](https://www.ai-expo.hu/) and at SZTAKI. For those who want to have some insight into the proof assistant languages like Coq, Lean, HOL4, I have found this tutorial on Isabelle/HOL, that is basic enough and at the same time detailed and concrete: 
https://bookdown.org/aleksander_mendoza_drosik/learn-isabelle/fundamentals-of-isabelle.html

**Comment by icsabai** [comment]
ChatGPT-4 is announced: https://openai.com/research/gpt-4

**Comment by icsabai** [comment]
__Chomsky vs. GPT__

Yesterday someone called my attention to this [article](https://www.nytimes.com/2023/03/08/opinion/noam-chomsky-chatgpt-ai.html) in which Noam Chomsky argues against the statement that large language models can reach real intelligence. I agree with some of the reasoning (like current version of LLMs lack "grounding", as the "wheat of chess" example nicely shows), but he had the following criticism:

_"For this reason, the predictions of machine learning systems will always be superficial and dubious. Because these programs cannot explain the rules of English syntax, for example, they may well predict, incorrectly, that “John is too stubborn to talk to” means that John is so stubborn that he will not talk to someone or other (rather than that he is too stubborn to be reasoned with)."_

It is dangerous to give concrete examples as they can be tested. And as an empiricist, I did so:

![image](https://user-images.githubusercontent.com/18164480/229167393-52b2e00f-1d9f-4f8c-90da-01e4737118f2.png)

Chomsky vs. GPT - 0:1

As the references at the bottom have shown, chatGPT could just look this up in the dictionary (as humans look up meaning of expressions in their internal dictionary).  Seems like Chomsky is making judgement without knowing too much about the workings of GPT, a typical bug in human intelligence.

**Comment by icsabai** [comment]
One open source GPT alternative that may be used for finetuning on special scientific/medical texts like the 1M covid papers (https://www.covid19dataportal.org/search/literature) I mentioned last time:
- https://www.databricks.com/blog/2023/04/12/dolly-first-open-commercially-viable-instruction-tuned-llm
- https://huggingface.co/databricks
